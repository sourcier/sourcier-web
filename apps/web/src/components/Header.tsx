import { Link } from 'gatsby';

export const Header = ({ siteTitle, siteDescription }) => (
  <Link to="/" className="md:col-span-3">
    <h1>{siteTitle}</h1>
    <p>{siteDescription}</p>
  </Link>
);
