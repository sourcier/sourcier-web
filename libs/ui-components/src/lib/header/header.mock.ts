export const brand = 'Sourcier';

export const nav = [
  { text: 'Home', href: '/' },
  { text: 'Link', href: '/link' },
];
