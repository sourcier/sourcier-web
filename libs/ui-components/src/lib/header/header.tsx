import clsx from 'clsx';
import { useState, useEffect } from 'react';
import { FaBars, FaTimes } from 'react-icons/fa';

export interface HeaderProps {
  brand: string;
  nav?: {
    text: string;
    href: string;
  }[];
}

export function Header({ brand, nav }: HeaderProps) {
  const [isVisible, setIsVisible] = useState(false);

  useEffect(() => {
    if (document && window) {
      if (isVisible) {
        document.body.style.position = 'fixed';
        document.body.style.top = `-${window.scrollY}px`;
      } else {
        const scrollY = document.body.style.top;
        document.body.style.position = '';
        document.body.style.top = '';
        window.scrollTo(0, parseInt(scrollY || '0') * -1);
      }
    }
  }, [isVisible]);

  return (
    <nav className="pt-4 bg-grey-900 text-grey-900-contrast">
      <div className="max-w-3xl px-4 mx-auto">
        <div className="flex justify-between pb-4">
          <div className="flex space-x-7">
            <div>
              {/* brand */}
              <a href="/" className="flex items-center px-2 py-4">
                <span
                  className="text-xl font-semibold"
                  data-testid="site-brand"
                >
                  {brand}
                </span>
              </a>
            </div>

            {/* nav items */}
            <div className="items-center hidden space-x-1 md:flex">
              {nav &&
                nav.map(({ text, href }, idx) => (
                  <a
                    href={href}
                    className="px-2 py-4 font-semibold hover:text-amber"
                    key={`nav-item-${idx}`}
                  >
                    {text}
                  </a>
                ))}
            </div>
          </div>

          {/* mobile menu button */}
          <div className="flex items-center md:hidden">
            <button
              className="outline-none text-amber"
              onClick={() => setIsVisible(!isVisible)}
            >
              {isVisible ? (
                <FaTimes className="w-6 h-6" />
              ) : (
                <FaBars className="w-6 h-6" />
              )}
            </button>
          </div>
        </div>
      </div>

      {/* mobile menu */}
      <div
        className={clsx(
          'h-screen text-center text-xl border-t-2 border-grey-800 pt-8',
          {
            hidden: !isVisible,
          }
        )}
      >
        <ul>
          {nav &&
            nav.map(({ text, href }, idx) => (
              <li key={`mobile-nav-item-${idx}`}>
                <a href={href} className="block px-2 py-4">
                  {text}
                </a>
              </li>
            ))}
        </ul>
      </div>
    </nav>
  );
}

export default Header;
