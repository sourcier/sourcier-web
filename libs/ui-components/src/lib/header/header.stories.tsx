import { Story, Meta } from '@storybook/react';

import { Header, HeaderProps } from './header';
import { brand, nav } from './header.mock';

export default {
  component: Header,
  title: 'Header',
} as Meta;

const Template: Story<HeaderProps> = (args) => <Header {...args} />;

export const Primary = Template.bind({});
Primary.args = {
  brand,
  nav,
};
