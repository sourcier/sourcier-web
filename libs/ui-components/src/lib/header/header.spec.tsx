import renderer from 'react-test-renderer';

import Header from './header';
import { brand, nav } from './header.mock';

describe('Header', () => {
  it('should render successfully', () => {
    const tree = renderer.create(<Header nav={nav} brand={brand} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
