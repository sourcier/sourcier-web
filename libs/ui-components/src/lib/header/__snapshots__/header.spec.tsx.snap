// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header should render successfully 1`] = `
<nav
  className="pt-4 bg-grey-900 text-grey-900-contrast"
>
  <div
    className="max-w-3xl px-4 mx-auto"
  >
    <div
      className="flex justify-between pb-4"
    >
      <div
        className="flex space-x-7"
      >
        <div>
          <a
            className="flex items-center px-2 py-4"
            href="/"
          >
            <span
              className="text-xl font-semibold"
              data-testid="site-brand"
            >
              Sourcier
            </span>
          </a>
        </div>
        <div
          className="items-center hidden space-x-1 md:flex"
        >
          <a
            className="px-2 py-4 font-semibold hover:text-amber"
            href="/"
          >
            Home
          </a>
          <a
            className="px-2 py-4 font-semibold hover:text-amber"
            href="/link"
          >
            Link
          </a>
        </div>
      </div>
      <div
        className="flex items-center md:hidden"
      >
        <button
          className="outline-none text-amber"
          onClick={[Function]}
        >
          <svg
            className="w-6 h-6"
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            strokeWidth="0"
            style={
              Object {
                "color": undefined,
              }
            }
            viewBox="0 0 448 512"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M16 132h416c8.837 0 16-7.163 16-16V76c0-8.837-7.163-16-16-16H16C7.163 60 0 67.163 0 76v40c0 8.837 7.163 16 16 16zm0 160h416c8.837 0 16-7.163 16-16v-40c0-8.837-7.163-16-16-16H16c-8.837 0-16 7.163-16 16v40c0 8.837 7.163 16 16 16zm0 160h416c8.837 0 16-7.163 16-16v-40c0-8.837-7.163-16-16-16H16c-8.837 0-16 7.163-16 16v40c0 8.837 7.163 16 16 16z"
            />
          </svg>
        </button>
      </div>
    </div>
  </div>
  <div
    className="h-screen text-center text-xl border-t-2 border-grey-800 pt-8 hidden"
  >
    <ul>
      <li>
        <a
          className="block px-2 py-4"
          href="/"
        >
          Home
        </a>
      </li>
      <li>
        <a
          className="block px-2 py-4"
          href="/link"
        >
          Link
        </a>
      </li>
    </ul>
  </div>
</nav>
`;
